packages/reactive-element/src/decorators/query-all.ts

```diff
@@ -59,13 +59,25 @@ let fragment: DocumentFragment;
  * @category Decorator
  */
 export function queryAll(selector: string): QueryAllDecorator {
-  return (() => {
-    return {
+  return ((
+    obj: object,
+    name: PropertyKey | ClassAccessorDecoratorContext<unknown, unknown>
+  ) => {
+    const descriptor = {
+      ...descriptorDefaults,
       get(this: ReactiveElement) {
         const container =
           this.renderRoot ?? (fragment ??= document.createDocumentFragment());
         return container.querySelectorAll(selector);
       },
     };
+    if (typeof name !== 'object' && extendedReflect.decorate) {
+      // If we're called as a legacy decorator, and Reflect.decorate is present
+      // then we have no guarantees that the returned descriptor will be
+      // defined on the class, so we must apply it directly ourselves.
```

Comment:
I don't fully understand why this is the case (I have a vague memory of you talking about it), but I do appreciate the comment stating this behavior.

Reply:
TypeScript experimental decorators defer to the nonstandard Reflect.decorate if it's present, and some (all?) Angular apps install a very ancient version that ignores the returned value